### PR TITLE
[AJ-1279] Add tests for SQL expressions for data type conversions

### PR DESCRIPTION
**Pull request by nawatts**
Following up on https://github.com/DataBiosphere/terra-workspace-data-service/pull/472#discussion_r1465451372...

This adds tests that show the SQL expression generated for each attribute data type conversion.

### CODE REVIEW

**Review by calypsomatic — APPROVED** [review]

**Review by jladieu — COMMENTED** [review]
Thanks for doing this!

I have some ideas about how to improve readability that I plan to express in the form of a PR based off this...ETA 10 minutes.

**Review by jladieu — APPROVED** [review]
Oops, I accidentally pushed my proposed changes up onto your branch as a new commit, instead of creating a proper PR for this.  Please treat these as optional and feel free to blow them away if you dislike.

But basically I was trying to get as many of the args declarations to fit onto one line as possible.  I thought this might make it easier to scan the test fixture, even though it doesn't help the most verbose cases.

File: service/src/test/java/org/databiosphere/workspacedataservice/dao/RecordDaoTest.java [hunk 1]
```diff
@@ -459,8 +457,7 @@ void testDeleteRelatedRecord() {
         instanceId,
         recordType,
         Collections.singletonList(testRecord),
-        new HashMap<>(
-            Map.of("foo", DataTypeMapping.STRING, "testRecordType", DataTypeMapping.RELATION)));
+        new HashMap<>(Map.of("foo", STRING, "testRecordType", RELATION)));
```
Comment (jladieu) [inline service/src/test/java/org/databiosphere/workspacedataservice/dao/RecordDaoTest.java:460]: This looks like existing code, so you don't need to change this, but I want to point out that you _shouldn't_ need to create a `HashMap` out of the result of `Map.of` to pass this through.

I'm trying this out on a branch to see if there's a reason why this was necessary.

Comment (nawatts) [reply service/src/test/java/org/databiosphere/workspacedataservice/dao/RecordDaoTest.java:460]: Looks like this was added in https://github.com/DataBiosphere/terra-workspace-data-service/pull/62.

Comment (jladieu) [reply service/src/test/java/org/databiosphere/workspacedataservice/dao/RecordDaoTest.java:460]: Pushed up https://github.com/DataBiosphere/terra-workspace-data-service/pull/478 for this.

**Pull request merged**